python/nav/eventengine/plugins/upgrade.py

```diff
@@ -0,0 +1,49 @@
+#
+# Copyright (C) 2022 Sikt
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License version 3 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.  You should have received a copy of the GNU General Public
+# License along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+"""software upgrade handler plugin"""
+
+from nav.eventengine.plugin import EventHandler
+from nav.eventengine.alerts import AlertGenerator
+
+
+class UpgradeHandler(EventHandler):
+    """Accepts deviceNotice events"""
+
+    handled_types = ('deviceNotice',)
+
+    def handle(self):
+
+        event = self.event
+
+        if event.state != event.STATE_STATELESS:
+            self._logger.info('Ignoring stateful deviceNotice event')
+        else:
+            self._post_alert(event)
+
+        event.delete()
+
+    def _post_alert(self, event):
+        alert = AlertGenerator(event)
+        if alert.alert_type not in (
+            "deviceHwUpgrade",
+            "deviceSwUpgrade",
+            "deviceFwUpgrade",
+        ):
+            self._logger.info('Ignoring deviceNotice event with alert type deviceError')
+            return
+
+        alert.history_vars = {"old_version": alert["old_version"]}
```

Comment:
Isn't there also a field for the new version? Or do we need to go back and add that in as well?

Reply:
Yes, but that is in saved in the device already, so we don't need to put that into the varmap

Reply:
@johannaengland Not quite. The *current* version is stored in the device entry. But an upgrade event from last year did not necessarily upgrade to the current version, but an older one. 

You could grok this by ordering all upgrade events by date and looking at the `old_version` attribute of the next event - until you reach the latest event, in which case you need to look at `device.*version` instead. But that would be unnecessarily complex. 

It would be much easier to store both `old_version` and `new_version` in each event. Then you can get the information you need from any single event.

Reply:
That is very true, I did not think about that. Thanks!